test/micro/org/openjdk/bench/java/lang/foreign/MemorySegmentVsBits.java

```diff
@@ -94,6 +103,25 @@ public void bitsEquivalent() {
             putLong(bytes, i * Long.BYTES, longs[i]);
         }
     }
+    @Benchmark
+    public void byteVarHandle() {
+        for (int i = 0; i < size; i++) {
+            LONG_ARRAY_VH.set(bytes, i * Long.BYTES, longs[i]);
```

Comment:
Is this offset multiplication correct? e.g. I would assume that var handles created with `byteArrayView` would take a logical index?

Reply:
Nevermind - that seems to be correct - it's equivalent to `ByteBuffer.getLong`

Reply:
Might be worth trying this benchmark with the VM flag: `-Djava.lang.invoke.VarHandle.VAR_HANDLE_GUARDS=false`, to see if VH guards are contributing to a major speedup here.

Reply:
> Is this offset multiplication correct? e.g. I would assume that var handles created with `byteArrayView` would take a logical index?

No the coordinate of those handles is always raw byte offset. It is a bit confusing in the documentation as it also tells you about unaligned accesses.

https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/lang/invoke/MethodHandles.html#byteArrayViewVarHandle(java.lang.Class,java.nio.ByteOrder) says:

"Produces a VarHandle giving access to elements of a byte[] array viewed as if it were a different primitive array type, such as int[] or long[]. The VarHandle's variable type is the component type of viewArrayClass and the list of coordinate types is (byte[], int), **where the int coordinate type corresponds to an argument that is an index into a byte[] array. The returned VarHandle accesses bytes at an index in a byte[] array, composing bytes to or from a value of the component type of viewArrayClass according to the given endianness.**"